protos/capture.proto

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+syntax = "proto3";
+
+message SchedulingSlice {
+  int32 pid = 1;
+  int32 tid = 2;
+  int32 core = 3;
+  uint64 in_timestamp_ns = 4;
+  uint64 out_timestamp_ns = 5;
+}
+
+message Callstack {
+  repeated uint64 pcs = 1;
+}
+
+message InternedCallstack {
+  uint64 key = 1;
+  Callstack intern = 2;
+}
+
+message CallstackSample {
+  int32 pid = 1;
+  int32 tid = 2;
+  oneof callstack_or_key {
+    Callstack callstack = 3;
+    uint64 callstack_key = 4;
+  }
+  uint64 timestamp_ns = 5;
+}
+
+message FunctionCall {
+  int32 pid = 1;
+  int32 tid = 2;
+  uint64 virtual_address = 3;
+  uint64 begin_timestamp_ns = 4;
+  uint64 end_timestamp_ns = 5;
+  int32 depth = 6;
+  uint64 return_value = 7;
+}
+
+message InternedString {
```

Comment:
I am trying to think of a better name but only StringWithKey comes to mind.

Reply:
I think I still prefer StringAndKey/CallstackAndKey - they are more readable. Interned needs to be decoded couple of times before you remember it.

Reply:
> Interned needs to be decoded couple of times before you remember it.

What do you mean with this? As I said this is standard nomenclature, I don't see a reason to change it for something less accurate.

Reply:
So is InternedSomething always means something with key?

Reply:
There has to be some id. In the context of Java/C# in the same process, this ends up being the reference itself and it's somewhat transparent, but the same concept applies.